src/overlays/actors/ovl_Door_Ana/z_door_ana.c

```diff
@@ -21,20 +24,166 @@ const ActorInit Door_Ana_InitVars = {
+    this->actor.shape.rot.y = this->actor.shape.rot.z = 0;
+
+    if ((grottoType == DOORANA_TYPE_UNK) || (grottoType == DOORANA_TYPE_HIDDEN)) {
+        if (grottoType == DOORANA_TYPE_HIDDEN) {
+            Collider_InitAndSetCylinder(globalCtx, &this->bombCollider, &this->actor, &sCylinderInit);
+
+        } else {
+            this->actor.flags |= 0x10; // always update 
+        }
+
+        Actor_SetScale(&this->actor, 0);
+        DoorAna_SetupAction(this, DoorAna_WaitClosed);
+
+    } else {
+        DoorAna_SetupAction(this, DoorAna_WaitOpen);
+    }
+
+    this->actor.targetMode = 0;
+}
+
+void DoorAna_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DoorAna* this = THIS;
+    s32 grottoType = GET_DOORANA_TYPE(this);
+
+    if (grottoType == DOORANA_TYPE_HIDDEN) {
+        Collider_DestroyCylinder(globalCtx, &this->bombCollider);
+    }
+}
+
+void DoorAna_WaitClosed(DoorAna* this, GlobalContext* globalCtx) {
+    s32 grottoIsOpen = 0;
+    u32 grottoType = GET_DOORANA_TYPE(this);
+
+    if ((grottoType) == DOORANA_TYPE_UNK) {
+        // in OOT decomp its marked as open with storms, but does not seem to open with storms in MM
+        if ((this->actor.xyzDistToPlayerSq < 40000.0f) && (func_800F13E8(globalCtx, 5) != 0)) {
+            grottoIsOpen = 1;
+            this->actor.flags &= ~0x10; // always update OFF
+        }
+
+    } else {
+        if ((this->bombCollider.base.acFlags & AC_HIT) ) { // bomb collision
+            grottoIsOpen = 1;
+            Collider_DestroyCylinder(globalCtx, &this->bombCollider);
+
+        } else {
+            Collider_UpdateCylinder(&this->actor, &this->bombCollider);
+            CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx , &this->bombCollider.base);
+        }
+    }
+
+    if (grottoIsOpen) {
+        this->actor.params &= ~DOORANA_TYPE_BITRANGE;
+        DoorAna_SetupAction(this, DoorAna_WaitOpen); // change update function to "waiting open"
+        play_sound(0x4802); // CORRECT_CHIME sfx
+    }
+
+    func_800B8C50(&this->actor, globalCtx);
+}
+
+void DoorAna_WaitOpen(DoorAna* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s32 dooranaType = GET_DOORANA_TYPE(this);
+    s8 pad[4];
+    s32 entranceIndex;
+    f32 yDist;
+
+    if (Math_StepToF( &this->actor.scale, 0.01f, 0.001f) != 0) { 
+        if ((this->actor.targetMode != 0) && (globalCtx->unk18875 == 0) && (globalCtx->unk18B4A == 0) &&
+          (player->stateFlags1 & 0x80000000) && (player->unkAE7 == 0)) {
+
+            if ((dooranaType) == DOORANA_TYPE_ADJACENT) {
+                // 300 uses scene exit addresses, not static grotto addresses, 
+                // eg. deku playground gets address in the NCT scene exit table, not in grotto table
+
+                entranceIndex = this->actor.params & 0x1F;
+                globalCtx->unk1887A = globalCtx->setupExitList[entranceIndex];
+                label_required_to_convince_compiler_to_not_use_delay_slot: ;
```

Comment:
This is almost certainly not necessary but I can understand leaving it in if it's too much trouble otherwise

Reply:
Sure looks required, doesn't match without it

![required label](https://user-images.githubusercontent.com/42048411/113954766-53823a00-97cf-11eb-949f-7f1ea35f3d25.png)

Reply:
It might be required as the code currently is, but I think the current thinking is that these fake labels are basically always fake, and that there is another solution 99.99% of the time

Reply:
I suggest a generic `lblUnk:;`